adadamp/_dist.py

```diff
@@ -57,13 +78,23 @@ def gradient(
     where `l` is the loss function for a single example.
 
     """
-    if idx is not None:
-        inputs = inputs[idx]
-        targets = targets[idx]
-    inputs, targets = inputs.to(device), targets.to(device)
+    # Dirty hack! This function should not be in this file.
+    #
+    # However, client.scatter(train_set) created problems with
+    # client.scatter(model)
+    # See https://github.com/dask/distributed/issues/3807 for more detail
+    train_set = _get_fashionmnist()
+
+    data_target = [train_set[i] for i in idx]
+    inputs = [d[0].reshape(-1, *d[0].size()) for d in data_target]
+    targets = [d[1] for d in data_target]
```

Comment:
Why is `data_target` defined and looped over twice? Why not loop over `train_set` twice?

Reply:
I thought it was so that we don't have to pay for the transform caused by `train_set[I]` twice by pulling each pt out once and putting it in a list. I may have misunderstood what these lines were doing though.

Reply:
Yup, that's exactly it. I poorly worded my comment; I should have said "add a comment explaining."